### ISSUE TITLE
PHP Fatal error:  Class phpbu\App\Backup\Target\Compression\Gzip contains 1 abstract method (composer)

### ISSUE THREAD

**Issue by atrandafir** [sebastianfeldmann/phpbu#384]
I'm running the one installed from composer on PHP 8.1 and I get this error:

```
/opt/plesk/php/8.1/bin/php /var/www/vhosts/example.com/webapp/vendor/bin/phpbu --configuration=/var/www/vhosts/example.com/webapp/common/config/phpbu.xml
phpbu 6.0.28 by Sebastian Feldmann and contributors.

Runtime:       PHP 8.1.31
Configuration: /var/www/vhosts/example.com/webapp/common/config/phpbu.xml

PHP Fatal error:  Class phpbu\App\Backup\Target\Compression\Gzip contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (SebastianFeldmann\Cli\Command::__toString) in /var/www/vhosts/momorentacar.com/webapp/vendor/phpbu/phpbu/src/Backup/Target/Compression                                                                                     /Gzip.php on line 15
```

I've downloaded it instead from https://phar.phpbu.de/phpbu.phar and ran with the same config and same PHP version and I get no error.

**Comment by sebastianfeldmann** [comment]
Oh damn! This is a breaking change I introduced in my own library :/

#epicfail

Will be fixed shortly

**Comment by sebastianfeldmann** [comment]
Should be fixed in `6.0.29`
sorry

**Comment by iainsaxon** [comment]
I'm working on an old project that still uses PHPBU 6.0.20 and PHP 7.4. When running the backup I get:

```
Fatal error: Class phpbu\App\Backup\Target\Compression\Bzip2 contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (SebastianFeldmann\Cli\Command::__toString) in /var/www/html/vendor/phpbu/phpbu/src/Backup/Target/Compression/Bzip2.php on line 15
```

I was able to resolve my issue by requiring `sebastianfeldmann/cli` 3.4.1 in my project's composer file as that was all that changed when updating my composer packages today.

Composer correctly skipped versions of PHPBU that were incompatible with PHP 7.4 however the `cli` package got a version bump to 3.4.2 that triggered the above error and I suspect its because there isn't a limit to what versions of CLI are listed as compatible with PHPBU.

Thanks again for PHPBU!
